client/src/contexts/SocketContext.tsx

```diff
@@ -10,23 +11,43 @@ export const useSocket = () => {
 
 export const SocketProvider = ({ children }: { children: React.ReactNode }) => {
   const [socket, setSocket] = useState<Socket | null>(null);
+  const locationContext = useLocation();
 
   useEffect(() => {
-    const fetchIP = async () => {
-      try {
-        const networkState = await Network.getNetworkStateAsync();
-        if (networkState.isConnected) {
-          const ip = await Network.getIpAddressAsync();
-          const socketIo = io(`http://${ip}:8080`); // Dynamically fetches and uses the IP address
-          setSocket(socketIo);
-        }
-      } catch (error) {
-        console.error('Could not get IP address:', error);
+    let isMounted = true;
+
+    const socketIo = io('http://{your-expo-ip}:8080'); // Hardcoded IP address
```

Comment:
This hardcoded value might be hard to find while setting up the project, could you please switch this to use `EXPO_PUBLIC_BACKEND_ADDRESS` from the .env?

And why did you remove the local IP retrieval? Was it not working?

Reply:
Yeah it wasn't working. It retrieved the devices IP, not the Expo IP :<